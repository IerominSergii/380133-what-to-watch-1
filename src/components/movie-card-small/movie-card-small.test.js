import React from "react";
import renderer from "react-test-renderer";
import MovieCardSmall from "./movie-card-small";

it(`should render MovieCardSmall properly`, () => {
  const id = 1;
  const src = `videoPath`;
  const imagePath = `imagePath`;
  const title = `title`;
  const activeMovieId = 1;
  const onMouseOver = jest.fn();
  const onMouseOut = jest.fn();

  const movieCardSmall = renderer
    .create(
        <MovieCardSmall
          id={id}
          src={src}
          imagePath={imagePath}
          title={title}
          activeMovieId={activeMovieId}
          onMouseOver={onMouseOver}
          onMouseOut={onMouseOut}
        />
    )
    .toJSON();
  console.log(movieCardSmall.debug());

  expect(movieCardSmall).toMatchSnapshot();
});
