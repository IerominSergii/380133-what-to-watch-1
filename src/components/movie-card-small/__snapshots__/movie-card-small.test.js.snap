// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieCardSmall properly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  id={1}
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <button
    className="small-movie-card__play-btn"
    onClick={[MockFunction]}
    type="button"
  >
    Play
  </button>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="title"
      height="175"
      src="img/imagePath"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      title
    </a>
  </h3>
</article>
`;
